helper/impls/helper-impl-110/src/main/java/com/linkedin/avroutil1/compatibility/avro110/FieldBuilder110.java

```diff
@@ -92,6 +93,34 @@ public FieldBuilder setOrder(Order order) {
     return this;
   }
 
+  @Override
+  public FieldBuilder addProp(String propName, String jsonLiteral) {
+    ObjectMapper objectMapper = new ObjectMapper();
+    try {
+      _props.put(propName, objectMapper.readTree(jsonLiteral));
+      return this;
+    } catch (JsonProcessingException e) {
+      throw new IllegalArgumentException("Failed to parse serialized json object: " + jsonLiteral, e);
+    }
+  }
+
+  @Override
+  public FieldBuilder addProps(Map<String, String> propNameToJsonObjectMap) {
+    for (Map.Entry<String, String> entry : propNameToJsonObjectMap.entrySet()) {
+      addProp(entry.getKey(), entry.getValue());
```

Comment:
nit - catch illegalArg exception, wrap it and state the prop name in your wrapped exception (otherwise folks will know the bad value, but not the key it was under)